### PR TITLE
webview, openLink: Get the AWS res file path and open it in browser.

### PR DESCRIPTION
**Pull request by jainkuniya**
Auth needs to be passed to access user_upload, or we can also access it
via appending `api_key` to the url. So append the `api_key` to the url,
and then get the redirected url (which is of AWS). And finally open it
in browser, so that user don't have to login to realm in browser.

So the external browser only sees the "expires in 1 minute" S3 URL for
accessing the file. And the `api_key` is also not exposed in the url via
accessing the file in browser.

https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/view.20uploaded.20files

### CODE REVIEW

File: src/message/messagesActions.js [hunk 1]
```diff
@@ -34,6 +34,24 @@ export const messageLinkPress = (href: string) => (dispatch: Dispatch, getState:
...
+      const srcPath = url.substring(auth.realm.length);
+      if (
+        srcPath.startsWith('/user_uploads/')
+        || srcPath.startsWith('/thumbnail?')
+        || srcPath.startsWith('/avatar/')
+      ) {
+        const delimiter = url.includes('?') ? '&' : '?';
+        url += `${delimiter}api_key=${auth.apiKey}`;
+
+        fetch(url).then(response => {
```
Comment (borisyankov) [inline src/message/messagesActions.js]: async + await > then

File: src/message/messagesActions.js [hunk 2]
```diff
@@ -34,6 +34,24 @@ export const messageLinkPress = (href: string) => (dispatch: Dispatch, getState:
...
+          openLink(response.url);
```
Comment (borisyankov) [inline src/message/messagesActions.js]: This is likely not enough. I don't thick all realms' resources are behind a redirect to URL AWS.

File: src/message/messagesActions.js (hunk 1 above)
Comment (jainkuniya) [reply src/message/messagesActions.js]: Done 👍

**Comment by chrisbobbe** [comment]
To clear a path forward on this important issue, I'm closing both this PR and #3495, with a solution outlined at https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/view.20uploaded.20files/near/819777 that will involve some backend work.

There, Greg lists these three requirements that the new solution fulfills, and offers an implementation plan:

```quote
The goal basically is to get
* (a) a plain old URL that can be passed to a browser (not a URL plus some cookies or other headers that need to be set)
* (b) that gets the user the file immediately, without requiring a fresh login
* (c) but doesn't include a secret with much wider powers (i.e. their API key), which would risk a nasty surprise if they innocently show someone else the file by sending them the link.
```

**Pull request closed without merging**